### DISCUSSION TITLE
Zod refine validation with various modes

### DISCUSSION THREAD

**Discussion by dvnrsn** [react-hook-form/react-hook-form#6742] [Q&A]
[codesandbox](https://codesandbox.io/s/dreamy-microservice-4nwru?file=/src/App.js)

Is there a way to run refine with various modes such as "onBlur" or "onChange". AFAICT it runs only on form submission.

This is not the case for basic object definition which emits errors according to specified mode.

Edit: Looks like trigger may need to be involved

https://github.com/react-hook-form/react-hook-form/discussions/4174

@bluebill1049 's comment adds clarity here.

> Sounds like dependant field validation, in hook form we focus on single input at a time, if you need to trigger other inputs' validation, you will need to use triggger api.

**Comment by bluebill1049** [comment]
I have been very busy at work and maintain this lib (lately with those TS issues), i hope i can get to this RFC soon: https://github.com/react-hook-form/react-hook-form/discussions/6472 this should resolve the issue.
